core/sql/sqlcomp/PrivMgrPrivileges.cpp

```diff
@@ -3182,34 +3042,107 @@ PrivStatus PrivMgrPrivileges::revokeColumnPrivileges(
 
    // Create list of ColumnReferences
    objectUsage.columnReferences = new std::vector<ColumnReference *>;
-   for (size_t i = 0; i < colPrivsToRevoke.size(); i++)
+   for (size_t i = 0; i < colPrivsToRevoke.entries(); i++)
    {
-      ColPrivEntry &colPrivToRevoke = colPrivsToRevoke[i];
-      ColPrivEntry *grantedColPriv = findColumnEntry(grantedColPrivs, colPrivToRevoke.getColumnOrdinal());
+      PrivMgrCoreDesc &colPrivToRevoke = colPrivsToRevoke[i];
+      PrivMgrCoreDesc *grantedColPriv = findColumnEntry(grantedColPrivs, colPrivToRevoke.getColumnOrdinal());
       if (grantedColPriv)
       {
-         ColumnReference *adjustedCol = new ColumnReference;
-         adjustedCol->columnOrdinal = colPrivToRevoke.getColumnOrdinal();
-         adjustedCol->originalPrivs = grantedColPriv->getPrivDesc();     
-         PrivMgrCoreDesc adjustedPrivs = grantedColPriv->getPrivDesc();
-         adjustedPrivs.AndNot(colPrivToRevoke.getPrivDesc());
-         adjustedCol->updatedPrivs = adjustedPrivs;
-         objectUsage.columnReferences->push_back(adjustedCol);
-      }
-   }
+        if (colPrivToRevoke.anyNotSet(*grantedColPriv))
+        {
+          // sanity check -> verify that privileges to revoke actually are set
+          // in the granted list 
+          for (size_t p = FIRST_DML_COL_PRIV; p <= LAST_DML_COL_PRIV; p++ )
+          {
+            PrivType type = (PrivType)p;
+
+            // If trying to revoke a privilege that is not granted or 
+            // if trying to revoke grant option that is not granted, report it
+            //if ((colPrivToRevoke.getPriv(type) && !grantedColPriv->getPriv(type)) ||
+            //    (!colPrivToRevoke.getPriv(type) && colPrivToRevoke.getWgo(type) && !grantedColPriv->getWgo(type)))
+            bool printWarning = false;
+            bool printWgo = false;
+            if (colPrivToRevoke.getPriv(type))
+            {
+              if ( !grantedColPriv->getPriv(type))
+                printWarning = true;
+            }
+            else
+            {
+              if (colPrivToRevoke.getWgo(type) && !grantedColPriv->getWgo(type))
+              {
+                printWarning = true;
+                printWgo = true;
```

Comment:
I didn't see printWgo used anywhere. It's only a problem though if there is some logic you intended but forgot.